pennylane/measurements/probs.py

```diff
@@ -139,33 +139,35 @@ def return_type(self):
     def numeric_type(self):
         return float
 
-    def shape(self, device=None):
+    def shape(self, config=None, len_wires=None):
         if qml.active_return():
-            return self._shape_new(device)
-        if device is None:
+            return self._shape_new(config=config, len_wires=len_wires)
+        if config is None:
             raise MeasurementShapeError(
-                "The device argument is required to obtain the shape of the measurement "
+                "The config argument is required to obtain the shape of the measurement "
                 f"{self.__class__.__name__}."
             )
         num_shot_elements = (
-            1 if device.shot_vector is None else sum(s.copies for s in device.shot_vector)
+            1 if config.shot_vector is None else sum(s.copies for s in config.shot_vector)
         )
-        len_wires = len(self.wires)
-        dim = self._get_num_basis_states(len_wires, device)
+        n_wires = len(self.wires)
+        dim = self._get_num_basis_states(n_wires, config)
+        # TODO: Change/fix this (maybe??)
 
         return (num_shot_elements, dim)
 
-    def _shape_new(self, device=None):
-        if device is None:
+    def _shape_new(self, config=None, len_wires=None):
+        if config is None:
             raise MeasurementShapeError(
-                "The device argument is required to obtain the shape of the measurement "
+                "The config argument is required to obtain the shape of the measurement "
                 f"{self.__class__.__name__}."
             )
         num_shot_elements = (
-            1 if device.shot_vector is None else sum(s.copies for s in device.shot_vector)
+            1 if config.shot_vector is None else sum(s.copies for s in config.shot_vector)
         )
-        len_wires = len(self.wires)
-        dim = self._get_num_basis_states(len_wires, device)
+        n_wires = len(self.wires)
+        dim = self._get_num_basis_states(n_wires, config)
+        # TODO: Change/fix this (maybe??)
```

Comment:
what's the issue here?

Reply:
`MeasurementProcess._get_num_basis_states` determines how many basis states there are depending on the number of wires as well as the cutoff of cv devices. `ExecutionConfig` would need a `cutoff` attribute for this to work properly, otherwise `MeasurementProcess._get_num_basis_states` will always return `2**num_wires` even when the number of dimensions is higher (whether for cv or qutrit devices)

Reply:
@albi3ro can confirm, but `cutoff` sounds like something that would go into the `device_options` attribute of our ExecutionConfig! I'd be ok to put the ugly type-checking logic in that private function (eg. `if isinstance(config, EC); cutoff = config.device_options.get("cutoff")`)